# Generated by Django 3.0.6 on 2020-07-07 07:31

import datetime
from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='AUTHORIZE_TYPE',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('kind', models.CharField(choices=[('Approved', 'Approved'), ('Rejected', 'Rejected'), ('Pending', 'Pending')], max_length=8)),
            ],
        ),
        migrations.CreateModel(
            name='EXCEPTION',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('state', models.CharField(choices=[('Approved', 'Approved'), ('Rejected', 'Rejected'), ('Pending', 'Pending')], default='Pending', max_length=8)),
                ('exception_type', models.IntegerField(null=True)),
                ('patch_id', models.TextField(null=True)),
                ('title', models.CharField(max_length=30)),
                ('justification', models.TextField()),
                ('action_plan', models.TextField()),
                ('exclude_date', models.DateTimeField(default=datetime.datetime.now)),
                ('content', models.TextField(null=True)),
                ('server_id', models.TextField(null=True)),
                ('client', models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.CreateModel(
            name='EXCEPTION_TYPE',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('kind', models.CharField(choices=[('Patch', 'Patch'), ('Server', 'Server')], max_length=8)),
            ],
        ),
        migrations.CreateModel(
            name='VALIDATE_EXCEPTION',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('state', models.CharField(choices=[('Approved', 'Approved'), ('Rejected', 'Rejected'), ('Pending', 'Pending')], default='Pending', max_length=8)),
                ('comment', models.TextField(blank=True, default='Pending')),
                ('time', models.DateTimeField(default=datetime.datetime.now)),
                ('approver', models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, related_name='approverValidate', to=settings.AUTH_USER_MODEL)),
                ('exception', models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, to='exception.EXCEPTION')),
            ],
        ),
    ]
